Controller/Component/CsvComponent.php

```diff
@@ -141,5 +146,26 @@ public function export($filename, $data, $options = array()) {
 			return false;
 		}
 	}
+	
+	public function setFileNameOut($name = 'export-data.csv')
+	{
+		return $this->filename_out = $name;
+	}
+
+	public function getFileNameOut()
+	{
+		return $this->filename_out;
+	}
+
+	public function sendHttpHeaders() 
+	{
+		header('Content-Encoding: UTF-8');
+		header('Content-Type: text/csv; charset=utf-8' );
+		header(sprintf( 'Content-Disposition: attachment; filename=%s', basename($this->getFileNameOut()) ) );
+		header('Content-Transfer-Encoding: binary');
+		header('Expires: 0');
+		header('Cache-Control: must-revalidate, post-check=0, pre-check=0');
+		header('Pragma: public');
+	}

```

Comment:
Instead of making `export()` more complex, why don't you add a `download()` method that calls `export()` internally and sets the headers and filename appropriately. This way you could take the download filename as an argument to `download()` and still pass `php://output` to the `export()` function.